src/components/subscriptionDetail/SubscriptionInstance.tsx

```diff
@@ -58,22 +91,36 @@ export default function SubscriptionInstance({
             )}
             {subscription_instance.label && <p className="label">{`Label: ${subscription_instance.label}`}</p>}
             <p className="label">{`Instance ID: ${subscription_instance.subscription_instance_id}`}</p>
+
             <table className="detail-block multiple-tbody">
                 <thead />
-                {mapSplitFields(subscription_instance.subscription_instance_id, value_fields, inUseBySubscriptions)}
+                {mapSplitFields(
+                    subscription_instance.subscription_instance_id,
+                    updatedValueFields,
+                    inUseBySubscriptions
+                )}
             </table>
-            {instance_fields
-                .flatMap((entry) => entry[1].map((value: any) => [entry[0], value]))
-                .map((entry: [string, ISubscriptionInstance], i: number) => (
-                    <SubscriptionInstance
-                        key={i}
-                        subscription_id={subscription_id}
-                        subscription_instance={entry[1]}
-                        field_name={field_name ? `${field_name}.${entry[0]}` : entry[0]}
-                        inUseBySubscriptions={inUseBySubscriptions}
-                        showRelatedBlocks={showRelatedBlocks}
-                    />
-                ))}
+
+            {shouldRenderInstanceFields &&
+                instance_fields
```

Comment:
move this into a function, to keep the template clean of logic.

Reply:
Done, introduced a mapping function